File: src/main/java/org/blacksun/graph/channel/HalfDuplexChannelFactory.java
package org.blacksun.graph.channel;

import org.blacksun.graph.node.GraphNode;
import org.jetbrains.annotations.NotNull;

public class HalfDuplexChannelFactory implements ChannelFactory {
    @Override
    public Channel createChannel(@NotNull GraphNode from, @NotNull GraphNode to, int weight, double errors) {
        return new HalfDuplexChannel(from, to, weight, errors);
    }

    @Override
    public String toString() {
        return "Half-duplex";
    }
}

File: src/main/java/org/blacksun/graph/channel/HalfDuplexChannelFactory.kt
package org.blacksun.graph.channel

import org.blacksun.graph.node.GraphNode

class HalfDuplexChannelFactory : ChannelFactory {
    override fun createChannel(from: GraphNode, to: GraphNode, weight: Int, errors: Double): Channel {
        return HalfDuplexChannel(from, to, weight, errors)
    }

    override fun toString(): String {
        return "Half-duplex"
    }
}
